src/LinkContainer.js

```diff
@@ -66,42 +48,63 @@ class LinkContainer extends React.Component {
     }
 
     if (
-      target ||
-      event.defaultPrevented ||
-      isModifiedEvent(event) ||
-      !isLeftClickEvent(event)
+      !event.defaultPrevented && // onClick prevented default
+      event.button === 0 && // ignore right clicks
+      !this.props.target && // let browser handle "target=_blank" etc.
+      !isModifiedEvent(event) // ignore clicks with modifier keys
     ) {
-      return;
-    }
+      event.preventDefault();
 
-    event.preventDefault();
+      const { router } = this.context;
+      const { replace, to } = this.props;
 
-    this.context.router[action](
-      createLocationDescriptor(to, query, hash, state)
-    );
-  };
+      if (replace) {
+        router.replace(to);
+      } else {
+        router.push(to);
+      }
+    }
+  }
 
   render() {
-    const { router } = this.context;
-    const { onlyActiveOnIndex, to, children, ...props } = this.props;
-
-    props.onClick = this.onClick;
-
-    // Ignore if rendered outside Router context; simplifies unit testing.
-    if (router) {
-      props.href = router.createHref(to);
+    const {
+      children,
+      replace, // eslint-disable-line no-unused-vars
+      to,
+      exact,
+      strict,
+      activeClassName,
+      className,
+      activeStyle,
+      style,
+      isActive: getIsActive,
+      ...props,
+    } = this.props;
 
-      if (props.active == null) {
-        props.active = router.isActive(to, onlyActiveOnIndex);
-      }
-    }
+    const href = this.context.router.createHref(
+      typeof to === 'string' ? { pathname: to } : to
+    );
 
-    return React.cloneElement(React.Children.only(children), props);
+    return (
+      <Route
```

Comment:
Based on [official example](https://reacttraining.com/react-router/examples/custom-link). `children` function is called even if current URL doesn't match. It's called with an object that contains a `match` and a `history` properties but `match` will be *null* if there was no match. Thus it becomes possible to find out if `activeClassName` and/or `activeStyle` should be applied to the container.